radio/src/gui/128x64/model_setup.cpp

```diff
@@ -921,6 +921,23 @@ void menuModelSetup(event_t event)
             }
           }
         }
+        else if (IS_MODULE_CROSSFIRE(moduleIdx)) {
+          lcdDrawTextAlignedLeft(y, INDENT "Baudrate");
+          lcdDrawNumber(MODEL_SETUP_2ND_COLUMN, y, CROSSFIRE_BAUDRATES[g_eeGeneral.mavbaud], attr|LEFT);
+          if (attr) {
+            g_eeGeneral.mavbaud = DIM(CROSSFIRE_BAUDRATES) - 1 - checkIncDecModel(event, DIM(CROSSFIRE_BAUDRATES) - 1 - g_eeGeneral.mavbaud, 0, DIM(CROSSFIRE_BAUDRATES) - 1);
+            if (checkIncDec_Ret) {
+              pauseMixerCalculations();
+              pausePulses();
+              EXTERNAL_MODULE_OFF();
+              CoTickDelay(10); // 20ms so that the pulses interrupt will reinit the frame rate
+              telemetryProtocol = 255; // force telemetry port + module reinitialization
+              EXTERNAL_MODULE_ON();
+              resumePulses();
+              resumeMixerCalculations();
+            }
```

Comment:
Messing with timers and so here seems very ugly. I would have have in telemetryWakeup check if the baudrate is still right as part of the checks if telemetry needs to reinitalised.